pkg/clamav/scanner.go

```diff
@@ -0,0 +1,86 @@
+package clamav
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	clamd "github.com/dutchcoders/go-clamd"
+	"github.com/fsouza/go-dockerclient"
+
+	"github.com/openshift/image-inspector/pkg/api"
+)
+
+type ClamScanner struct {
+	// Socket is the location of the clamav socket.
+	Socket string
+}
+
+var _ api.Scanner = &ClamScanner{}
+
+func NewScanner(socket string) api.Scanner {
+	return &ClamScanner{
+		Socket: socket,
+	}
+}
+
+// Scan will scan the image
+func (s *ClamScanner) Scan(path string, image *docker.Image) ([]api.Result, error) {
+	scanner := clamd.NewClamd(s.Socket)
+
+	versionChan, err := scanner.Version()
+	if err != nil {
+		return nil, err
+	}
+	scannerVersion := <-versionChan
+
+	// Useful for debugging
+	scanStarted := time.Now()
+	defer func() {
+		log.Printf("ClamAv scan took %ds", int64(time.Since(scanStarted).Seconds()))
+	}()
+	resultChan, err := scanner.ContScanFile(path)
```

Comment:
ContScanFile is the directory one I believe.   That is what I expected to probably be using when we get past the WIP.